### DISCUSSION TITLE
Using Import names from Console?

### DISCUSSION THREAD

**Discussion by bakerwy** [fumitoh/modelx#155] [Q&A] [answered]
Hi,

I frequently use "Import names" from modelx GUI widgets because it makes modeling works very easy.
(It doesn't require space name for variables).

I wonder if I can use this feature in the Console.

Thanks in advance!

**Comment by fumitoh** [comment]
Are you using "Import names" on a single cells or on a space to import all the names in the space?

For the former, can you just do like below?

```python

foo = AModel.ASpace.foo

```

You may also want to change the auto-completion setting to make it work [as explained here](https://github.com/fumitoh/modelx/issues/128#issue-2355395189)

**Reply by bakerwy** [reply]
I usually use Import Name for a space not a cell. So it seems there is no simple way to do that right?

**Reply by fumitoh** [reply, marked as answer]
No, but I think it's possible to introduce a convenience method to do that:

```python
AModel.ASpace.export_names()
```
Let me think about it.

**Reply by bakerwy** [reply]
Hi, 
I wonder this thread is on going or it has some technical issue.
This will be great functions for heavy users.

**Reply by fumitoh** [reply]
After thinking a little bit more about the design, I feel it's cleaner to implement this feature as a modelx API function, not as a Space method.

```python
mx.export_names(AModel.ASpace)
```

The implementation would be something like this:

```python
def export_names(space, module='__main__'):
    ns = sys.modules[module]
    for k, v in space.cells.items():
        setattr(ns, k, v)
    ...
```
So, when you call this function from an interactive Python console, which is named `__main__` most of the time, you can just omit the `module` parameter.
I can add this quickly.

**Reply by fumitoh** [reply]
This has been implemented as `export_memebers` : https://docs.modelx.io/en/latest/releases/relnotes_v0_29_1.html
